File: web/src/figures/fig_1_9.ts
import Plotly from 'plotly.js-dist-min';
import { Rnl, constants } from '../common/hydrogenic';
import { slider } from '../common/ui';

const controls = document.getElementById('controls')!;
const plot = document.getElementById('plot')!;
const caption = document.getElementById('caption')!;
caption.textContent = 'FIGURE 1.9 Radial wavefunctions R_n1 for 2p and 3p.';

const zSlider = slider(controls, 'Z', 1, 3, 1);

const compute = () => {
  const Z = parseInt(zSlider.value, 10);
  const xs: number[] = [];
  const r2: number[] = [];
  const r3: number[] = [];
  for (let i = 0; i <= 300; i++) {
    const x = (30 * i) / 300;
    const r = (x * constants.a0) / Z;
    xs.push(x);
    r2.push(Rnl(2, 1, r, Z));
    r3.push(Rnl(3, 1, r, Z));
  }
  Plotly.newPlot(plot, [
    { x: xs, y: r2, mode: 'lines', name: '2p', line: { color: '#9b59b6' } },
    { x: xs, y: r3, mode: 'lines', name: '3p', line: { color: '#16a085' } }
  ], {
    margin: { l: 60, r: 20, t: 20, b: 40 },
    xaxis: { title: 'x = Zr/a₀' },
    yaxis: { title: 'R_n1' },
    legend: { orientation: 'h' }
  }, { displayModeBar: false });
};

zSlider.addEventListener('input', compute);
compute();

File: pluto/fig_1_9.jl
### A Pluto.jl notebook ###
# v0.19.0

using PlutoUI
using WGLMakie
include("common/hydrogenic.jl")
using .Hydrogenic

md"""# Figure 1.9
Radial wavefunctions R_n1 for 2p and 3p."""

@bind Z Slider(1:3, default=1)

xs = range(0, 30; length=400)
rs = [x*a0/Z for x in xs]
R2 = [R_nl(2,1,r; Z=Z) for r in rs]
R3 = [R_nl(3,1,r; Z=Z) for r in rs]

fig = Figure(resolution=(800,400))
ax = Axis(fig[1,1], xlabel="x = Zr/a0", ylabel="R_n1")
lines!(ax, xs, R2, color=:purple, label="2p")
lines!(ax, xs, R3, color=:teal, label="3p")
axislegend(ax)
fig
